src/main/java/org/folio/service/RequestExpirationService.java

```diff
@@ -119,35 +119,34 @@ private static Future<List<Request>> getExpiredRequests(AsyncResult<SQLConnectio
         .collect(toList()));
   }
 
-  private static Future<List<Request>> getOpenRequestsByItemIds(
-    AsyncResult<SQLConnection> conn, Vertx vertx, String tenant, Set<String> itemIds) {
-
-    PostgresClient pgClient = PostgresClient.getInstance(vertx, tenant);
-    Promise<RowSet<Row>> promise = Promise.promise();
+  private Future<List<Request>> getOpenRequestsByIdFields(AsyncResult<SQLConnection> conn,
+    Set<String> idFields) {
 
-    if (itemIds.isEmpty()) {
+    if (idFields.isEmpty()) {
       return succeededFuture(emptyList());
     }
 
-    Set<String> quotedItemIds = itemIds.stream()
+    Set<String> quotedFieldIds = idFields.stream()
       .map(id -> format("'%s'", id))
       .collect(toSet());
-
     String where = format("WHERE " +
         "(jsonb->>'status' = '%1$s' OR " +
         "jsonb->>'status' = '%2$s' OR " +
         "jsonb->>'status' = '%3$s' OR " +
         "jsonb->>'status' = '%4$s') AND " +
-        "jsonb->>'itemId' IN (%5$s) " +
+        "jsonb->>'%5$s' IN (%6$s) " +
         "ORDER BY jsonb->>'position' ASC",
 
       OPEN_NOT_YET_FILLED.value(),
       OPEN_AWAITING_PICKUP.value(),
       OPEN_AWAITING_DELIVERY.value(),
       OPEN_IN_TRANSIT.value(),
-      String.join(",", quotedItemIds));
+      idFieldName,
+      String.join(",", quotedFieldIds));
 
-    String fullTableName = format("%s.%s", PostgresClient.convertToPsqlStandard(tenant), REQUEST_TABLE);
+    Promise<RowSet<Row>> promise = Promise.promise();
```

Comment:
Methods `getExpiredRequests` and `getOpenRequestsByIdFields` share this part:
```
Promise<RowSet<Row>> promise = Promise.promise();
    String fullTableName = format("%s.%s", PostgresClient.convertToPsqlStandard(
      pgClient.getTenantId()), REQUEST_TABLE);
    String sql = format("SELECT jsonb FROM %s %s", fullTableName, where);
    pgClient.select(conn, sql, promise);

    return promise.future()
      .map(rs -> rowSetToStream(rs)
        .map(row -> row.get(JsonObject.class, row.getColumnIndex(JSONB_COLUMN)))
        .map(json -> json.mapTo(Request.class))
        .collect(toList()));
```
Can we extract it into a method?